src/Protocol/Protocol_1_8.cpp

```diff
@@ -1532,12 +1533,14 @@ void cProtocol_1_8_0::SendUpdateBlockEntity(cBlockEntity & a_BlockEntity)
 	Byte Action = 0;
 	switch (a_BlockEntity.GetBlockType())
 	{
-		case E_BLOCK_MOB_SPAWNER:   Action = 1; break;  // Update mob spawner spinny mob thing
-		case E_BLOCK_COMMAND_BLOCK: Action = 2; break;  // Update command block text
-		case E_BLOCK_BEACON:        Action = 3; break;  // Update beacon entity
-		case E_BLOCK_HEAD:          Action = 4; break;  // Update Mobhead entity
-		case E_BLOCK_FLOWER_POT:    Action = 5; break;  // Update flower pot
-		case E_BLOCK_BED:           Action = 11; break;  // Update bed color
+		case E_BLOCK_MOB_SPAWNER:       Action = 1; break;  // Update mob spawner spinny mob thing
+		case E_BLOCK_COMMAND_BLOCK:     Action = 2; break;  // Update command block text
+		case E_BLOCK_BEACON:            Action = 3; break;  // Update beacon entity
+		case E_BLOCK_HEAD:              Action = 4; break;  // Update Mobhead entity
+		case E_BLOCK_FLOWER_POT:        Action = 5; break;  // Update flower pot
+		case E_BLOCK_BED:               Action = 11; break;  // Update bed color
+		case E_BLOCK_ENCHANTMENT_TABLE: Action = 0; break;
```

Comment:
What meaning does a zero action have here? Wiki.vg is not very illuminating.

Reply:
it's a bit of abuse or ugly workaround

I needed a way to send the block entities to a player

Reply:
Hm, I'm not a protocol expert, so maybe anyone else can comment here; but if it works then it's fine :+1:

Reply:
Or is there another way?

You may send block entity during the chunk sending BUT this isn't implemented in cuberite.

Reply:
Ah, that makes sense more completely now. You should add a comment here explaining that this is a workaround, with TODO to fix it after adding block entities sending during chunk sending.

Reply:
Only 1.9.4+'s Chunk Data has any provision to send tile entities. I just tried with 1.8.9-1.9.4 and these versions don't need you to actually send this packet for the enchantment table floating book to appear (haven't tested end portals).

Reply:
I developed with 1.11. and this version required me sending those.

Reply:
The other block entitires which are not listed there do have a block action making them visible to the client e.g. Ender Chest. BUT the enchanting table and the ender portal frame don't have that.

Reply:
Anyway add the TODO, we can remove the workaround when sending block entities in chunk data is implemented.

Reply:
Weren't these normal blocks that later became block entities? If that's the case you shouldn't send any packet in the versions where they aren't entities. I can't find a reference for when that was though.

Reply:
We can handle that in chunkdataserializer, I don't think Action = 0 has any ill effects as a workaround

Reply:
Chunk serialisation is irrelevant. You aren't going to resend the entire chunk every time a block entity gets placed.

Reply:
We can drop all changes to SendUpdateBlockEntity in all versions when serialisation is fixed. These workarounds are just to declare the entity during initial chunk send so blocks aren't invisible; the client creates tile entities itself during normal block placement.

Reply:
exept for the end portal... If you place a enchanting table you see the book till logout. But the end portal blocks are invisible if they spawn

Reply:
On 1.12 SetBlock an end portal during gameplay shows up fine, no invisibility

Reply:
On relog it doesn't

Reply:
Sure, that's what we're talking about fixing with chunk sending in the future :)

Reply:
> On 1.12 SetBlock an end portal during gameplay shows up fine, no invisibility

This got me a bit confused. I thought you talked about the release implementation

Reply:
>We can drop all changes to SendUpdateBlockEntity in all versions when serialisation is fixed.

Plugins can also change block entity properties and there's no way the client could possibly know that.

Reply:
Indeed but from what I can see this packet doesn't have any Action corresponding to updating enchanting tables or end portals (1.16.3). By drop changes I meant the special Action = 0 added in this PR needn't stay forever.

Reply:
resend whole chunk then? :/

Reply:
I dunno, we can figure that out later.

Reply:
so we agree this works for now and the one rewriting the SehndChunk code adds the block entites?